### PR TITLE
add watchdog feature, support restart child process throught -r param…

### PR DESCRIPTION
**Pull request by hubugui**
please help review, thanks!

### CODE REVIEW

File: chik/chik.cpp [hunk 1]
```diff
@@ -90,7 +101,13 @@ int main(int argc, char* argv[]) {
 		}
 
 		// This will cause windows to spawn a "cmd.exe /c" process
-		return system(command.c_str());
+		if (restart_child) {
+			while (1) {
+				system(command.c_str());
+			}
+		} else
```
Comment (thomas-kl) [inline chik/chik.cpp]: Can you add also braces "{}" for else to have common code-style?

File: chik/chik.cpp [hunk 2]
```diff
@@ -72,9 +72,20 @@ int main(int argc, char* argv[]) {
 
 	if (argc > 1) {
 		string command;
+		int i;
+		int restart_child = 0;
```
Comment (thomas-kl) [inline chik/chik.cpp]: Can you use bool instead of int here?

File: chik/chik.cpp [hunk 3]
```diff
@@ -72,9 +72,20 @@ int main(int argc, char* argv[]) {
 
 	if (argc > 1) {
 		string command;
+		int i;
```
Comment (thomas-kl) [inline chik/chik.cpp]: Init with "1", just to have a good default in case of later code changes?

**Review by thomas-kl — CHANGES_REQUESTED** [review]
Cool submissions, thanks!

In addition to the comments, can you also add the "-r" plus a description to the help text at the start of the main?

**Comment by thomas-kl** [comment]
Hey! I merged the changes to master, thanks!

**Comment by thomas-kl** [comment]
Now also released -> v1.2

**Comment by hubugui** [comment]
Hi, Sorry for I did not follow github recently.

Thanks for your merged!

**Pull request merged**